### ISSUE TITLE
DDS BC2_UNORM file loading issue

### ISSUE THREAD

**Issue by ddelion** [NVIDIAGameWorks/Falcor#9]
I've tried to load DDS file with BC2_UNORM format and without mips. This scenario crashes the engine.

There is the message from DX12 debug layer:
```
D3D12 ERROR: ID3D12Device::CreateCommittedResource: A texture that has D3D12_RESOURCE_FLAG_ALLOW_RENDER_TARGET set in D3D12_RESOURCE_DESC::Flags must be created with a format that either can be used as a render target, or cast to a format that could be used as a render target. FeatureLevel is D3D_FEATURE_LEVEL_12_1. D3D12_RESOURCE_DESC::Format is BC2_UNORM. [ STATE_CREATION ERROR #599: CREATERESOURCE_INVALIDMISCFLAGS]
```
What I've found is that the updateBindFlags function in Texture.cpp adds render target flag to this texture even though it is intended to be shader resource only.

**Comment by ddelion** [comment]
It turns out that this file has mips, but DDSD_MIPMAPCOUNT flag was not written by the original file creator. 

According to this page https://msdn.microsoft.com/en-us/library/windows/desktop/bb943982(v=vs.85).aspx 

> Note  When you write .dds files, you should set the DDSD_CAPS and DDSD_PIXELFORMAT flags, and for mipmapped textures you should also set the DDSD_MIPMAPCOUNT flag. However, when you read a .dds file, you should not rely on the DDSD_CAPS, DDSD_PIXELFORMAT, and DDSD_MIPMAPCOUNT flags being set because some writers of such a file might not set these flags.

**Comment by kyaoNV** [comment]
I am working on some fixes regarding loading DDS files with BCx formats. Can I get a link to the image you were talking about to test with?

**Comment by ddelion** [comment]
You can get Sponza from this project https://github.com/SaschaWillems/VulkanSponza.

**Comment by kyaoNV** [comment]
Fixed in 64eab29
